File: src/main/java/com/back/domain/member/member/dto/MemberWithUsernameDto.java
package com.back.domain.member.member.dto;

import com.back.domain.member.member.entity.Member;

import java.time.LocalDateTime;

public record MemberWithUsernameDto(
        int id,
        LocalDateTime createDate,
        LocalDateTime modifyDate,
        String name,
        String username,
        Double score,
        String profileImgUrl
) {
    public MemberWithUsernameDto(int id, LocalDateTime createDate, LocalDateTime modifyDate, String name, String username, Double score,  String profileImgUrl) {
        this.id = id;
        this.createDate = createDate;
        this.modifyDate = modifyDate;
        this.name = name;
        this.username = username;
        this.score = score;
        this.profileImgUrl = profileImgUrl;
    }

    public MemberWithUsernameDto(Member member) {
        this(
                member.getId(),
                member.getCreateDate(),
                member.getModifyDate(),
                member.getNickname(),
                member.getUsername(),
                member.getReputation().getScore(),
                member.getProfileImgUrl()
        );
    }
}

File: src/main/java/com/back/domain/member/member/dto/MemberWithUsernameDto.kt
package com.back.domain.member.member.dto

import com.back.domain.member.member.entity.Member
import java.time.LocalDateTime

@JvmRecord
data class MemberWithUsernameDto(
    val id: Int,
    val createDate: LocalDateTime?,
    val modifyDate: LocalDateTime?,
    val name: String?,
    val username: String?,
    val profileImgUrl: String?,
    val score: Double?
) {
    constructor(member: Member) : this(
        member.id,
        member.createDate,
        member.modifyDate,
        member.nickname,
        member.username,
        member.profileImgUrl,
        member.reputation.score
    )
}